redash_toolbelt/examples/find-table-names.py

```diff
@@ -0,0 +1,92 @@
+import itertools, json, re
+import click
+from redash_toolbelt import Redash
+
+
+# This regex captures three groups:
+#
+#   0. A FROM or JOIN statement
+#   1. The whitespace character between FROM/JOIN and table name
+#   2. The table name
+PATTERN = re.compile(r"(FROM|JOIN)( )([^\s\(\)]+)", flags=re.IGNORECASE)
+
+
+def find_table_names(url, key, data_source_id):
+
+    client = Redash(url, key)
+
+    schema_tables = [
+        token.get("name")
+        for token in client._get(f"api/data_sources/{data_source_id}/schema")
+        .json()
+        .get("schema", [])
+    ]
+
+    queries = [
+        i
+        for i in client.paginate(client.queries)
+        if i.get("data_source_id", None) == int(data_source_id)
+    ]
+
+    tables_by_qry = {
+        i["id"]: [
+            re_groups[2]
+            for re_groups in re.findall(PATTERN, i["query"])
+            if re_groups[2] in schema_tables or len(schema_tables) == 0
+        ]
+        for i in queries
+        if re.search(PATTERN, i["query"])
+    }
+
+    return tables_by_qry
+
+
+def print_summary(tables_by_qry):
+    """Builds a summary showing table names and count of queries that reference it."""
+
+    summary = {
+        t: sum([1 for tables in tables_by_qry.values() if t in tables])
+        for t in itertools.chain(*tables_by_qry.values())
+    }
+
+    align = max([len(t) for t in summary.keys()])
+
+    print("\n")
+    print(f"{'table':>{align}} | {'number of queries':>17}")
```

Comment:
![giphy](https://user-images.githubusercontent.com/289488/79155894-88d19300-7dda-11ea-9e60-a72785bdd888.gif)

Reply:
Took me forever to learn this specification language. I'm unstoppable now.